internal/app/handlers/middlewarehandler/gzipcompressor.go

```diff
@@ -0,0 +1,69 @@
+package middlewarehandler
+
+import (
+	"github.com/northmule/shorturl/internal/app/logger"
+	"github.com/northmule/shorturl/internal/app/services/compressor"
+	"net/http"
+	"strings"
+)
+
+func MiddlewareGzipCompressor(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
+
+		expectedContentTypes := map[string]bool{
+			"application/json":   true,
+			"text/html":          true,
+			"application/x-gzip": true,
+		}
+
+		requestContentType := request.Header.Values("Content-type")
+		contentTypeIsSupported := false
+		for _, contentType := range requestContentType {
+			if _, ok := expectedContentTypes[contentType]; ok {
+				contentTypeIsSupported = true
+				break
+			}
+		}
+		if !contentTypeIsSupported {
+			logger.Log.Sugar().Infof("Ответ сервера не сжимается, для Content-type не поддерживается сжатие")
+			next.ServeHTTP(response, request)
+			return
+		}
```

Comment:
Я бы советовал тут, как и далее, использовать метод `Get()` для получения значения хедера. Так можно достаточно сильно упростить и сократить код.

```suggestion
		requestContentType := request.Header.Get("Content-Type")
		if _, ok := expectedContentTypes[requestContentType]; !ok {
			logger.Log.Sugar().Infof("Ответ сервера не сжимается, для Content-type не поддерживается сжатие")
			next.ServeHTTP(response, request)
			return
		}
```